clients/TypeScript/packages/client/src/ChainSync/ChainSyncClient.ts

```diff
@@ -62,12 +63,15 @@ export const createChainSyncClient = async (
         } else {
           throw new UnknownResultError(response.result)
         }
-      }, 1)
+      }
+      const responseHandler = options?.sequential !== false ?
```

Comment:
The response handler is set to the underlying message handler function if configured for non-sequential processing.